crd-generator/api-v2/src/main/java/io/fabric8/crdv2/generator/CRDInfo.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.crdv2.generator;
+
+public class CRDInfo {
```

Comment:
This is a no-go for QOSDK: this class is missing the `getDependentClassNames` support we need.

Reply:
I can add this back, but just to double check - is there a quarkus utility for detecting the transitive graph of a given class?

Reply:
Also - if this is needed, is it okay to just provide the dependent classes across everything passed into the crd generator? The ResolvingContext.uriToJacksonSchema values will contain all of the introspected classes, but it's not tracked by CRD.

Reply:
> I can add this back, but just to double check - is there a quarkus utility for detecting the transitive graph of a given class?

I'm not sure if there's such a thing… I'll ask but, iirc, there wasn't one when I implemented this feature.

Reply:
> > I can add this back, but just to double check - is there a quarkus utility for detecting the transitive graph of a given class?
> 
> I'm not sure if there's such a thing… I'll ask but, iirc, there wasn't one when I implemented this feature.

So I checked, and, no, there still isn't such a thing in Quarkus

Reply:
> Also - if this is needed, is it okay to just provide the dependent classes across everything passed into the crd generator? The ResolvingContext.uriToJacksonSchema values will contain all of the introspected classes, but it's not tracked by CRD.

I'm not sure, I would need to look into things in deeper details. Obviously, this would probably mean more work on my side. Additionally, I'd rather not expose these internals to outside code, if possible. The `getDependentClassNames` method was implementation-agnostic so I think it would be useful to restore the functionality if possible.

Reply:
> So I checked, and, no, there still isn't such a thing in Quarkus

Ok, and do you need it tracked per crd, or is across everything good enough?

Reply:
@metacosm check now, it should map to what you were using before.

Reply:
> @metacosm check now, it should map to what you were using before.

It doesn't work completely as expected. In particular, the previous version was overeager and returned too many things (like the `Group` annotation) but the new version doesn't return enough, in particular when there's a class hierarchy being involved (e.g. if a CR extends an intermediate class instead of extending CustomResource directly, the new version won't provide the intermediate class in `getDependentClassNames` even though changes to that class could potentially impact the CRD).

Reply:
@metacosm check now. It's not filtering java level stuff, but it will get the interfaces and super classes as well.

Reply:
Yep, this seems to work nicely now. Will now do a full CI run with this PR to check.